refactor: replaced java.util.logging.Logger with System.Logger

MedTahiri opened the pull request:
This PR replaces all usages of java.util.logging.Logger with the more modern and flexible System.Logger API introduced in Java 9.

Ran ./gradlew test to verify no regression.
All tests pass successfully.

MedTahiri committed:
Refactor: replaced java.util.logging.Logger with System.Logger

cklellie commented on casual/casual-caller-http-client/src/main/java/se/laz/casual/http/HttpClient.java:
```diff
@@ -44,7 +43,7 @@ public ServiceReturn<CasualBuffer> request(URI uri, CasualBuffer buffer)
         Objects.requireNonNull(buffer, "buffer can not be null");
         MediaType mediaType = CasualBufferTypeConverter.convert(CasualBufferType.unmarshall(buffer.getType()));
-        LOG.finest(() -> "issuing http request to " + uri);
+        LOG.log(System.Logger.Level.TRACE,() -> "issuing http request to " + uri);
```
Please use a static import for the different Level.

cklellie replied:
One of the reasons we are moving over to System.Logger is to be able to use DEBUG rather than finest.
Therefore we would like to replace `finest( ...)` calls with `log( DEBUG,...)`, rather than TRACE.

If you could update these accordingly that would be greatly appreciated.

cklellie commented on casual/casual-caller/src/main/java/se/laz/casual/connection/caller/ConnectionFactoryEntry.java:
```diff
@@ -71,13 +70,13 @@ public void validate()
             // Failure to connect during validation should automatically invalidate ConnectionFactoryEntry
             valid = false;
-            LOG.log(Level.WARNING, e, ()->"Failed validation of CasualConnection with jndiName=" + connectionFactoryProducer.getJndiName() + ", received error: " + e.getMessage());
+            LOG.log(System.Logger.Level.WARNING, ()->"Failed validation of CasualConnection with jndiName=" + connectionFactoryProducer.getJndiName() + ", received error: " + e.getMessage());
```
The exception `e`, is no longer passed into the log call. Please add this back.

There are a few other locations in the PR where this has happen, if you could update them all please.

cklellie reviewed (changes requested):
Hello, and welcome!

Thank you very much for your PR, greatly appreciated.

I have reviewed your changes and added a few comments for some fixes and changes.

Additionally, every commit to `dev`, we consider to be a release, so we need to update the version prior to merging. We follow semver for versioning, so if you could please bump the patch version by changing the version in `versions.gradle` from `3.3.2` to `3.3.3`. 

Thanks again :)

MedTahiri committed:
fix(logging): replace finest with log(DEBUG) and restore exception parameter in log calls

- Updated all LOG.finest(...) calls to LOG.log(DEBUG, ...)
- Added back exception parameter 'e' in warning log calls
- Bumped version from 3.3.2 to 3.3.3 in versions.gradle
- Added static imports for Logger.Level constants

MedTahiri commented:
Hello,

I have made the requested fixes:
- Replaced all finest(...) calls with log(DEBUG, ...)
- Restored the exception parameter in all warning log calls
- Bumped the patch version from 3.3.2 to 3.3.3
- Added static imports for Logger.Level constants

Please let me know if there is anything else to update.
Thanks for your review!

cklellie commented on casual/casual-caller/src/main/java/se/laz/casual/connection/caller/TransactionLess.java:
```diff
@@ -62,17 +63,17 @@ public Optional<DiscoveryReturn> discover(ConnectionFactoryEntry connectionFacto
-          LOG.warning(() -> "failed domain discovery for: " + connectionFactoryEntry + " -> " + e);
-          LOG.warning(() -> "services: " + cachedItems.get(CacheType.SERVICE) + " queues: " + cachedItems.get(CacheType.QUEUE));
+          LOG.log(WARNING,() -> "failed domain discovery for: " + connectionFactoryEntry + " -> " + e,e);
+          LOG.log(WARNING,() -> "services: " + cachedItems.get(CacheType.SERVICE) + " queues: " + cachedItems.get(CacheType.QUEUE),e);
```
`e`  should not be on both, as it will result in two log entries of the same stacktrace. Please remove from this second log call.

cklellie commented on casual/casual-caller-http-client/src/main/java/se/laz/casual/http/HttpClient.java:
```diff
@@ -19,13 +19,14 @@
 import java.util.Objects;
-import java.util.logging.Logger;
+
+import static java.lang.System.Logger.Level.*;
```
We prefer not to use wildcard imports.

Please change to a specific import rather than wildcard `*`. 

`import static java.lang.System.Logger.Level.DEBUG;` in this case.

Update accordingly in all files please.

cklellie reviewed (changes requested):
Thank you for the updates, just 2 small changes please.
